### PR TITLE
Issue #9 - Converted endpoints to return POJOs

### PR DESCRIPTION
**Pull request by mpeng3**
Resolves Issue #9.
Converted all endpoints to return POJOs instead of Response objects, where possible.  Also wrote new tests to validate that endpoints are behaving as expected.

### CODE REVIEW

File: UrlShortener/src/test/java/matthewgroves/urlShortener/resources/UrlShortenerResourceTest.java [hunk 1]
```diff
@@ -68,4 +69,92 @@ public void redirectLinkFound() throws URISyntaxException {
...
+	
+	@Test
+	public void apiLinkNotFound() throws URISyntaxException {
+		// when
+		Throwable thrown = Assertions.catchThrowable(() -> {
+			resource.getExpandedUrl("10");
+		});
+		
+		// then
+		assertThat(thrown).isInstanceOf(NotFoundException.class);
```
Comment (StdVectorBool) [inline UrlShortener/src/test/java/matthewgroves/urlShortener/resources/UrlShortenerResourceTest.java:81]: Consider @Test(expected = NotFoundException.class) in future for brevity.

File: UrlShortener/src/main/java/matthewgroves/urlShortener/resources/UrlShortenerResource.java [hunk 2]
```diff
@@ -81,20 +112,24 @@ public Response expandUrl(@PathParam("id") String id) {
...
-				
-				ShortenedUrl sUrl = new ShortenedUrl(id, fullUrl, "http://" + config.getHostname() + ":"
-						+ config.getRuntimePort() + "/shrtn/" + Long.toString(id, 36));
-				return Response.ok().entity(sUrl).build();
-			}
-		} catch (Exception e) {
-			return Response.serverError().entity(e).build();
-		}
+	public ShortenedUrl addShortenedUrl(String fullUrl) {
+		if (fullUrl == null || fullUrl.length() == 0 || fullUrl.length() > 2048)
```
Comment (StdVectorBool) [inline UrlShortener/src/main/java/matthewgroves/urlShortener/resources/UrlShortenerResource.java:116]: FYI: Dropwizard has a validation layer too http://www.dropwizard.io/0.9.0/docs/manual/validation.html#validations.

So you could write:
`@NotEmpty @Length(max=2048) String fullUrl`
but then you wouldn't be able to test the class standalone.

**Comment by StdVectorBool** [comment]
Added very minor FYIs, but patch looks good.  Merge!

File: UrlShortener/src/test/java/matthewgroves/urlShortener/resources/UrlShortenerResourceTest.java (hunk 1 above)
Comment (mpeng3) [inline UrlShortener/src/test/java/matthewgroves/urlShortener/resources/UrlShortenerResourceTest.java:81]: Done.  That is much cleaner!

**Pull request merged**